include/igl/demangle.h

```diff
@@ -0,0 +1,78 @@
+// This file is part of libigl, a simple c++ geometry processing library.
+//
+// Copyright (C) 2019 Jeremie Dumas <jeremie.dumas@ens-lyon.org>
+//
+// This Source Code Form is subject to the terms of the Mozilla Public License
+// v. 2.0. If a copy of the MPL was not distributed with this file, You can
+// obtain one at http://mozilla.org/MPL/2.0/.
+#ifndef IGL_DEMANGLE_H
+#define IGL_DEMANGLE_H
+
+#include <string>
+#include <typeinfo>
+#include <type_traits> // std::is_same<T1,T2>
+
+#ifdef _MSC_VER
+#else // clang or gcc
+#include <cxxabi.h>
+#include <cstdlib>
+#include <algorithm> // std::remove
+#endif // clang or gcc branch of _MSC_VER
+
+namespace igl
+{
+
+#ifdef _MSC_VER
+  //! Demangles the type encoded in a string
+  /*! @internal */
+  inline std::string demangle(std::string mangledName)
+  {
+    std::string prefix = "class "; // MSVC does "class myClass" see: https://docs.microsoft.com/en-us/cpp/cpp/typeid-operator
```

Comment:
Not sure if this works with nested class names. I would use an auxiliary function to replace all instances of one string by another, e.g. [this](https://stackoverflow.com/questions/5343190/how-do-i-replace-all-instances-of-a-string-with-another-string).

Reply:
Thanks! done with 8d286fe3e3d4ae17dd4a01104b2deeb31be7aeae